src/Http/Routing.Abstractions/src/IOutboundParameterTransformer.cs

```diff
@@ -14,6 +14,6 @@ public interface IOutboundParameterTransformer : IParameterPolicy
         /// </summary>
         /// <param name="value">The route value to transform.</param>
         /// <returns>The transformed value.</returns>
-        string TransformOutbound(object value);
+        string? TransformOutbound(object? value);
```

Comment:
In DfaMatcherBuilder the returned value from the method has ToString called on it. DfaMatcherBuilder should validate the value is not null.